src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorDataSource.cs

```diff
@@ -108,8 +108,31 @@ public override IEnumerable<DataSourceObjectMetadata> GetChildObjects(DataSource
 
             if (parentMetadata.MetadataType == DataSourceMetadataType.Cluster && includeSizeDetails)
             {
-                var child = _nodes[parentMetadata.Urn].FirstOrDefault();
-                return child == null ? Enumerable.Empty<DataSourceObjectMetadata>() : _nodes[child.Urn];
+                var tableGroups = _nodes[parentMetadata.Urn];
```

Comment:
This code does not look specific to the dashboard. When we get children of the cluster in the Object explorer, shouldn't we return the table/function groups, rather than list of tables.

Reply:
The `includeSizeDetails` is only used for the Dashboard. It doesn't do a great job of describing that. I can rename it if you want.

Reply:
Can't figure that out by reading the code. Can you please add a comment above the if statement about the case this code covers.